File: packages/test-equivalence-matchers/src/boundaries/nodes/strings/Base64Encoded.ts
import { AnyMatcherOrData, BASE64_ENCODED_TYPE } from '../../../entities';
import { AnyMatcher } from '../base';

/**
 * Transformation matcher that matches a base64 encoded version of the given string or string matcher
 *
 * WARNING: Since many strings are accidentally decodable as base64, this matcher is
 * best combined with a more restrictive string matcher (eg `StringifiedJson`).
 */
export class Base64Encoded extends AnyMatcher {
  /** @internal */
  readonly '_case:matcher:type': typeof BASE64_ENCODED_TYPE;

  /** @internal */
  readonly '_case:matcher:child': AnyMatcherOrData;

  /** @internal */
  readonly '_case:matcher:resolvesTo' = 'string';

  /**
   * WARNING: Since many strings are accidentally decodable as base64, this matcher is
   * best combined with a more restrictive string matcher (eg `StringifiedJson`).
   *
   * @param child - The string or string matcher that would match the decoded string
   */
  constructor(child: AnyMatcherOrData) {
    super(BASE64_ENCODED_TYPE);
    this['_case:matcher:child'] = child;
  }

  /**
   * For non-TypeScript implementations (see [AnyMatcher.toJSON()](#\@case-contract-testing/test-equivalence-matchers.AnyMatcher.toJSON))
   *
   * @privateRemarks
   * This comment and the implementation is boilerplate on all matchers to avoid
   * outputting duplicate unimportant documentation on all matcher classes of
   * the docs. Only modify this comment or the implementation via search and replace.
   */
  override toJSON(): unknown {
    return super.toJSON();
  }
}

File: packages/dsl-java/src/main/java/io/contract_testing/contractcase/dsl/matchers/strings/Base64Encoded.java
package io.contract_testing.contractcase.dsl.matchers.strings;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonProperty;
import io.contract_testing.contractcase.dsl.ContractCaseDsl;
import io.contract_testing.contractcase.dsl.DslMatcher;
import jakarta.annotation.Generated;
import java.lang.Object;
import java.lang.String;
import lombok.Builder;
import lombok.Getter;
import lombok.Getter;
import lombok.Getter;
import org.jetbrains.annotations.NotNull;

/**
 * Matches a base64 encoded version of the given string or string matcher.<p>WARNING: Since many strings are accidentally decodable as base64, this matcher is
 * best combined with a more restrictive string matcher (eg {@code StringifiedJson}).
 */
@Generated("@contract-case/case-definition-generator")
@ContractCaseDsl
public class Base64Encoded implements DslMatcher {

  /**
   * ContractCase's internal type for this element
   */
  @Getter
  @JsonProperty("_case:matcher:type")
  private final String type;

  /**
   * Constant parameter resolvesTo
   */
  @Getter
  @JsonProperty("_case:matcher:resolvesTo")
  private final String resolvesTo = "string";

  /**
   * The string or string matcher to encode.
   */
  @Getter
  @JsonProperty("_case:matcher:child")
  private final Object child;

  @Builder
  public Base64Encoded(@NotNull final Object child) {
    this.type = "_case:Base64Encoded";
    this.child = child;
  }
}
